packages/storage/amplify_storage_s3_ios/ios/Classes/FlutterPrefixResolver.swift

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import Foundation
+import Amplify
+import Flutter
+import AmplifyPlugins
+import AWSMobileClient
+import amplify_core
+
+
+class FlutterPrefixResolver : AWSS3PluginPrefixResolver{
+    
+    var channel: FlutterMethodChannel
+    
+    init(flutterChannel: FlutterMethodChannel){
+        channel = flutterChannel
+    }
+    
+    func resolvePrefix(for accessLevel: StorageAccessLevel,
+                       targetIdentityId: String?,
+                       completion: @escaping (Result<String, StorageError>) -> Void) {
+        
+        
+        var accessLevelString : String
+        
+        switch accessLevel{
+        case StorageAccessLevel.guest: accessLevelString = "public"
```

Comment:
I think we should use `StorageAccessLevel.guest. serviceAccessPrefix` to get the prefix string instead of hardcoding it here.

Reply:
`StorageAccessLevel.guest.serviceAccessPrefix` [returns `"public"`](https://github.com/aws-amplify/amplify-ios/blob/c6d870a837c7765439fc06a31ba2aa1472913f05/AmplifyPlugins/Storage/AWSS3StoragePlugin/Support/Utils/StorageRequestUtils%2BGetter.swift#L64).